### PR TITLE
feat: Add uniswapv3-base

### PR DESCRIPTION
**Pull request by Eric-Warehime**
Add uniswapv3-base and some basic configuration around it.

### CODE REVIEW

**Review by aljo242 — APPROVED** [review]
lgtm @Eric-Warehime can we run a test deployment?

File: providers/apis/defi/uniswapv3/utils.go [hunk 1]
```diff
@@ -96,6 +97,44 @@ var (
...
+		Timeout:          1000 * time.Millisecond,
+		Interval:         2000 * time.Millisecond,
+		ReconnectTimeout: 2000 * time.Millisecond,
+		MaxQueries:       1,
+		URL:              "https://mainnet.base.org",
+	}
+
+	// DefaultBaseMarketConfig is the default market configuration for Uniswap V3. Specifically
+	// this is for Base mainnet.
+	DefaultBaseMarketConfig = types.CurrencyPairsToProviderTickers{
```
Comment (g3ndo9kar9) [inline providers/apis/defi/uniswapv3/utils.go]: Can we remove the default market config? My plan is to deprecate all of these + use our static marketmaps for testing

File: cmd/constants/markets.go [hunk 2]
```diff
@@ -5161,6 +5163,45 @@ var (
...
+						Metadata_JSON: uniswapv3.PoolConfig{
+							// REF: https://app.uniswap.org/explore/pools/base/0xBA3F945812a83471d709BCe9C3CA699A19FB46f7
+							Address:       "0xBA3F945812a83471d709BCe9C3CA699A19FB46f7",
+							BaseDecimals:  18,
+							QuoteDecimals: 18,
+							Invert:        false,
+						}.MustToJSON(),
+					},
+				},
+			},
```
Comment (davidterpay) [inline cmd/constants/markets.go:5202]: This looks like its supporting two different assets?

**Comment by davidterpay** [comment]
Can you update the dYdX provider to accept feeds from base? Also have we decided on the format of what the ticker will look like in their gov proposals?

**Review by aljo242 — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -17,8 +17,9 @@ APP_TOML ?= $(HOMEDIR)/config/app.toml
 CONFIG_TOML ?= $(HOMEDIR)/config/config.toml
 COVER_FILE ?= cover.out
 BENCHMARK_ITERS ?= 10
-USE_CORE_MARKETS ?= true
+USE_CORE_MARKETS ?= false
 USE_RAYDIUM_MARKETS ?= false
+USE_UNISWAPV3_BASE_MARKETS ?= true
```
Comment (davidterpay) [inline Makefile:22]: nit: Would either default all of these to true or false.

**Review by davidterpay — APPROVED** [review]

File: cmd/constants/markets.go [hunk 4]
```diff
@@ -5161,6 +5161,37 @@ var (
 		}
 	}
 	  `
+
+	// UniswapV3BaseMarketMap is used to initialize the Base market map. This only includes
+	// the markets that are supported by uniswapv3 on Base.
+	UniswapV3BaseMarketMap mmtypes.MarketMap
+
+	// UniswapV3BaseMarketMapJSON is the JSON representation of UniswapV3BaseMarketMap.
+	UniswapV3BaseMarketMapJSON = `
```
Comment (g3ndo9kar9) [inline cmd/constants/markets.go:5170]: nit: is this a sufficient # of markets from base to give us confidence in the provider's operation?

File: providers/apis/defi/uniswapv3/utils.go [hunk 5]
```diff
@@ -84,15 +88,30 @@ func (pc PoolConfig) MustToJSON() string {
...
-	Timeout:          1000 * time.Millisecond,
-	Interval:         2000 * time.Millisecond,
-	ReconnectTimeout: 2000 * time.Millisecond,
-	MaxQueries:       1,
-	Endpoints:        []config.Endpoint{{URL: URL}},
-}
+var (
+	// DefaultETHAPIConfig is the default configuration for the Uniswap API. Specifically this is for
+	// Ethereum mainnet.
+	DefaultETHAPIConfig = config.APIConfig{
```
Comment (g3ndo9kar9) [inline providers/apis/defi/uniswapv3/utils.go:94]: nit: could change provider default name from `BaseName` to smth that doesn't contain `Base`

**Review by g3ndo9kar9 — APPROVED** [review]
lgtm

File: cmd/constants/markets.go (hunk 4 above)
Comment (Eric-Warehime) [reply cmd/constants/markets.go:5170]: Added some more :D

**Pull request merged**